src/reader.rs

```diff
@@ -0,0 +1,144 @@
+use crate::{
+    wtf16::{get_single_surrogate_pair_code_point, is_surrogate_code_point},
+    Wtf16,
+};
+
+pub type CodePoint = u32;
```

Comment:
A little arbitrary that this is defined/exported from here (probably in general the shape of the "public API" of the crate is kind of random/likely incomplete)